src/ty/ty_check.rs

```diff
@@ -16,6 +16,10 @@ pub fn expect_ty(expected: Ty, found: Ty) -> TyResult<Ty> {
     }
 }
 
+fn expect_ty2(expected: Ty, ty1: Ty, ty2: Ty) -> TyResult<Ty> {
```

Comment:
Hmmm This makes me think we might as well write a macro for this. Something like `ensure_ty!(expected, ty1, ty2, ...)`

Reply:
Or if performance isn't a major concern a iterator based method would also work. But since I needed just one more checked parameter I didn't generalized it further. If we add more complex (ternary operators anyone?) checks we can still decide then.

Reply:
my only complain with an iterator based method is the hassle of wrapping the types you want to check in an iterable structure such as a vector or a slice:
```rust
ensure_ty(Ty::Bool, vec![ty1, ty2])?;
```
With a single type is even worse because you end up with
```rust
ensure_ty(Ty::Bool, vec![ty])?;
```
So we'd have to keep the old `expect_ty` for one type and have `ensure_ty` for several. Which isn't great.  I think that the macro is an easy way to have this variable argument function without bothering with wrapping your arguments to do so.

Reply:
Another thing is that it is possible to provide an empty iterator/slice. Should I (try to) add the macro now?

Reply:
it would be cool. For now it's enough to just match two or three comma separated arguments.